sycl/source/detail/graph_impl.cpp

```diff
@@ -343,6 +343,9 @@ void exec_graph_impl::createCommandBuffers(sycl::device Device) {
 }
 
 exec_graph_impl::~exec_graph_impl() {
+  WriteLock Lock(MMutex);
```

Comment:
I wonder if this is actually ever necessary. We would generally have a problem if some other function was waiting for this lock to be released before it starts working on this object, as it would get access when it had been destroyed.

If on the contrary we have to wait here for another function to finish using the object, I would assume we could as easily have hit case 1 above. Is that assumption correct?

Reply:
You're right.
Indeed, the way exec_graph_impl is used and the C++ semantic prevent concurrent access the destructor of this object. 
We thus removed this lock.